BizHawk.Emulation.Common/Interfaces/Services/IStatable.cs

```diff
@@ -20,17 +21,55 @@ public interface IBinaryStateable
 	public interface IStatable : IBinaryStateable, IEmulatorService
 	{
 		/// <summary>
-		/// Gets a value indicating whether the core would rather give a binary savestate than a text one. Both must function regardless
+		/// save state binary to a byte buffer
 		/// </summary>
-		bool BinarySaveStatesPreferred { get; }
+		/// <returns>you may NOT modify this.  if you call SaveStateBinary() again with the same core, the old data MAY be overwritten.</returns>
+		byte[] SaveStateBinary();
+	}
 
+	/// <summary>
+	/// Allows a core to opt into text savestates.
+	/// If a core does not implement this interface, a default implementation
+	/// will be used that doesn't yield anything useful in the states, just binary as text
+	/// </summary>
+	public interface ITextStatable : IStatable
+	{
 		void SaveStateText(TextWriter writer);
+
 		void LoadStateText(TextReader reader);
+	}
 
-		/// <summary>
-		/// save state binary to a byte buffer
-		/// </summary>
-		/// <returns>you may NOT modify this.  if you call SaveStateBinary() again with the same core, the old data MAY be overwritten.</returns>
-		byte[] SaveStateBinary();
+	public static class StatableExtensions
+	{
+		// Note: if/when we go .net core a much better way to achieve this
+		// will be default interface implementation, and remove the ITextStatable interface
+		public static void SaveStateText(this IStatable core, TextWriter writer)
+		{
+			if (core is ITextStatable textCore)
+			{
+				textCore.SaveStateText(writer);
+			}
+
+			var temp = core.SaveStateBinary();
+			temp.SaveAsHexFast(writer);
+		}
+
+		public static void LoadStateText(this IStatable core, TextReader reader)
+		{
+			if (core is ITextStatable textCore)
+			{
+				textCore.LoadStateText(reader);
+			}
+
+			string hex = reader.ReadLine();
+			if (hex != null)
+			{
+				var state = new byte[hex.Length / 2];

```

Comment:
I know most cores don't pool their buffers for this, but given this one thing will be used in lots of different places, might be nice to pool buffers here

Reply:
I like the idea, and now we have a singular place to do it, I think that's for another PR down the road though